connector/src/main/scala/com/microsoft/kusto/spark/utils/ContainerProvider.scala

```diff
@@ -57,17 +60,26 @@ class ContainerProvider(
       .build
   }
 
-  def getContainer: ContainerAndSas = {
+  def getContainer(maybeIngestionStorageParams: Option[Array[IngestionStorageParameters]] = None)
+      : ContainerAndSas = {
     // Refresh if storageExpiryMinutes have passed since last refresh for this cluster as SAS should be valid for at least 120 minutes
     val now = Instant.now(Clock.systemUTC())
     val secondsElapsed =
       now.getEpochSecond - lastRefresh.getEpochSecond // get the seconds between now and last refresh
-    if (storageUris.isEmpty ||
-      secondsElapsed > cacheExpirySeconds /* If the cache has elapsed , refresh */ ) {
-      refresh()
-    } else {
-      roundRobinIdx = (roundRobinIdx + 1) % storageUris.size
-      storageUris(roundRobinIdx)
+    val isCacheExpired = secondsElapsed > cacheExpirySeconds
+    maybeIngestionStorageParams match {
+      case Some(ingestionStorageParams) =>
+        processContainerResults(
```

Comment:
here you call processContainerResults on cache hits as well as misses .
lastRefresh is updated to now() and you will not refresh the key